dts/bindings/timer/atmel,sam-tc.yaml

```diff
@@ -21,6 +21,41 @@ properties:
       description: peripheral ID
       required: true
 
+    channel:
+      type: int
+      required: false
+      description: |
+        Timer / Counter channel to use, channel 0 is the default.
+        Valid range: 0 - 2
+
+    clk:
+      type: string
+      required: true
+      enum:
+        - "PCK6"
+        - "MCK_8"
+        - "MCK_32"
+        - "MCK_128"
+        - "SLCK"
```

Comment:
Clocks don't align with all SoC:
SAM4L uses different  Generic Clock instances at TIMER_CLOCK1 for each TC instance. The clock uses BUS clock and divisors are different.
SAM3X/4S/4E uses MCK_2 at TIMER_CLOCK1
SAME7x/V7x uses PCK6 at TIMER_CLOCK1

My suggestion is use TCK_1, TCK_2, .. TCK_5 instead. With that, we can resolve in code and have only one driver.

Reply:
It is up to you, I don't have preference. When I comment about TCK_1 .. I mean a shortcut for TIMER_CLOCK1.